snuba/replacer.py

```diff
@@ -53,7 +60,10 @@ def flush_batch(self, batch: Sequence[Replacement]) -> None:
             if count == 0:
                 continue
 
-            self.__replacer_processor.pre_replacement(replacement, count)
+            need_optimize = (
+                self.__replacer_processor.pre_replacement(replacement, count)
```

Comment:
This is also running optimize in situations other than when the flag is passed, is that intended?

Reply:
Which cases are you referring to ?

Reply:
Ignore this, might have read this incorrectly earlier